keps/sig-node/20200619-provisioning-resources-with-numa-topology.md

```diff
@@ -0,0 +1,240 @@
+---
+title: Provisioning the Node Resources With NUMA Topology Information
+authors:
+  - "@AlexeyPerevalov"
+  - "@swatisehgal"
+owning-sig: sig-node
+participating-sigs:
+  - sig-node
+  - sig-scheduling
+reviewers:
+  - "@dchen1107"
+  - "@derekwaynecarr"
+approvers:
+  - "@dchen1107"
+  - "@derekwaynecarr"
+creation-date: 2020-06-19
+last-updated: 2020-06-19
+status: implementable
+see-also:
+  - "/keps/sig-scheduling/20200612-deducted-topology-manager.md"
+---
+
+# Exposure Node Resources With NUMA Topology Information
+
+## Table of Contents
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+- [Design Details](#design-details)
+  - [Design based on CRI](#cri)
+  - [Design based on podresources interface of the kubelet](#podresources)
+  - [API](#api)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Alternatives](#alternatives)
+  - [Annotation approach](#annotation-approach)
+  - [NUMA specification in ResourceName](#numa-in-resourcename)
+<!-- /toc -->
+
+## Summary
+
+Cluster which contains several nodes with NUMA topology and
+enabled TopologyManager feature gate is not rare now. In such cluster
+could be a situation when TopologyManager's admit handler on the kubelet
+side refuses pod launching since pod's resources request doesn't sutisfy
+selected TopologyManager policy, in this case pod should be rescheduled.
+And so it can get stuck in the rescheduling cycle.
+To avoid such kind of problem scheduler should choose a node considering topology
+of the node and TopologyManager policy on the worker node.
+
+## Motivation
+
+For the sake of scheduling which is topology aware, resources with topology
+information should be provisioned.
+This KEP describes how it would be implemented.
+
+### Goals
+
+Provisioning resources with topology information.
+
+### Non-Goals
+
+ - modification of any public API
+ - improving and as a result modification of the TopologyManager and its policies
+
+## Proposal
+
+Introduce standalone daemon which will be run on each node in the cluster as daemonset. It should
+collect resources for running pods and provide it with CRD.
+
+## Design Details
+
+The design consists of part which describes how datum collected
+and how it was provided.
+
+Resources used by the pod could be obtained by podresources interface
+of the kubelet and by CRI from container runtime.
+
+To calculate available resources need to know all resources
```

Comment:
i think there is a tension here with runtime classes.

also keep in mind that adoption of the kube-reserved versus system-reserved cgroups is low.  most deployers still organize kube as a part of system-reserved budget.